invenio_records_lom/utils/vcard.py

```diff
@@ -0,0 +1,135 @@
+
+from collections.abc import Iterable
+from dataclasses import dataclass
+
+
+@dataclass
+class VCardMaker:
+    """Utility for making vcards version 4.0 (see `RFC 6350 <https://www.rfc-editor.org/rfc/rfc6350>`_ for the spec).
+
+    Defaults for arguments follow the spec.
+
+    :param bool final_line_break: whether to include a line-break after the final line
+    :param str line_break_str: which string to use for line-breaks
+    :param str | None output_encoding: output will be a ``bytes``-object encoded with this encoding, set to ``None`` to output a ``str``-object instead
+    """
+
+    final_line_break: bool = True
+    line_break_str: str = "\r\n"
+    output_encoding: str | None = "utf-8"
+
+    @staticmethod
+    def escape_noncompound_property(noncompound_property: str) -> str:
+        r"""Escape a vcard-property.
+
+        Escapes ``\``, ``\n``, and ``,``.
+        This differs from escaping compound properties in that ``;`` is not escaped.
+        """
+        return (
+            noncompound_property.replace("\\", "\\\\")
+            .replace(",", "\\,")
+            .replace("\n", "\\n")
+        )
+
+    @staticmethod
+    def utf_8_len(s: str) -> int:
+        """Return length of `s` in number of octets `s`'s "utf-8"-encoding has."""
+        return len(s.encode(encoding="utf-8"))
+
+    def line_wrap(self, unwrapped_vcard: str) -> str:
+        """Wrap lines as per vcard-spec.
+
+        - line-length in "utf-8"-encoding may not be longer than 75 octets (excluding newline-characters)
+        - can't break apart multi-octet-sequences when line-wrapping
+        - continued lines start with a space-character
+        """
+        input_lines = [line.lstrip() for line in unwrapped_vcard.splitlines()]
+        output_lines = []
+        for input_line in input_lines:
+            start = 0  # index where currently built line starts
+            prefix = ""  # prefix of currently built line, space for continued lines, counts towards line-length
+            length = 0  # length of currently built line, in number of "utf-8"-octets in its "utf-8" encoding
+            idx = 0  # index running through `input_line`
+            while idx < len(input_line):
+                char = input_line[idx]
+                length += self.utf_8_len(char)
+                if length > 75:
+                    # `char` would push currently built line over limit
+                    output_lines.append(prefix + input_line[start:idx])
+                    start = idx
+                    prefix = " "
+                    length = self.utf_8_len(prefix)
+                else:
+                    idx += 1
+            # handle rest of `input_line`
+            output_lines.append(prefix + input_line[start:])
+
+        return self.line_break_str.join(output_lines) + (
+            self.line_break_str if self.final_line_break else ""
+        )
+
+    def make_vcard(self, fn: str | Iterable[str]) -> str | bytes:
```

Comment:
readability: what does `fn` stands for? I would interpret it as a function, but that makes no sense with the defined types

Reply:
is `Names of arguments are the lower-cased version of properties defined by the spec.` defined by the spec or by you? I would prefer the written out way so fn -> formatted_name